### PR TITLE
Add stateful buttons on forms

### PR DESCRIPTION
**Pull request by zainab-amir**
[VAN-123](https://openedx.atlassian.net/browse/VAN-123)

Added stateful button component from paragon to
- LoginPage
- RegistrationPage
- ForgetPasswordPage
- ResetPasswordPage

ezgif com-gif-maker

### CODE REVIEW

File: src/forgot-password/ForgotPasswordPage.jsx [hunk 1]
```diff
@@ -79,12 +80,17 @@ const ForgotPasswordPage = (props) => {
               </p>
               <LoginHelpLinks page="forgot-password" />
             </div>
-            <Button
+            <StatefulButton
+              type="submit"
               className="btn-primary submit"
+              state={status}
+              labels={{
+                null: intl.formatMessage(messages['logisration.forgot.password.page.submit.button']),
```
Comment (waheedahmed) [inline src/forgot-password/ForgotPasswordPage.jsx:88]: If the label is the same, you don't need to specify it for each state. Specifying only `default` would work here.

File: src/logistration/LoginPage.jsx [hunk 2]
```diff
@@ -196,20 +203,24 @@ class LoginPage extends React.Component {
...
                 <LoginHelpLinks page="login" />
               </div>
-              <Button
+              <StatefulButton
+                type="submit"
                 className="btn-primary submit"
+                state={submitState}
+                labels={{
+                  default: intl.formatMessage(messages['logistration.sign.in.button']),
+                  pending: intl.formatMessage(messages['logistration.sign.in.button']),
```
Comment (waheedahmed) [inline src/logistration/LoginPage.jsx:212]: Same here, remove the label for pending state.

File: src/logistration/RegistrationPage.jsx [hunk 3]
```diff
@@ -295,12 +295,16 @@ class RegistrationPage extends React.Component {
...
             <span>By creating an account, you agree to the <a href="https://www.edx.org/edx-terms-service">Terms of Service and Honor Code</a> and you acknowledge that edX and each Member process your personal data in accordance with the <a href="https://www.edx.org/edx-privacy-policy">Privacy Policy</a>.</span>
-            <Button
-              className="btn-primary mt-4 submit"
+            <StatefulButton
+              type="submit"
+              className="btn-primary submit mt-4"
+              state={submitState}
+              labels={{
+                default: intl.formatMessage(messages['logistration.create.account.button']),
+                pending: intl.formatMessage(messages['logistration.create.account.button']),
```
Comment (waheedahmed) [inline src/logistration/RegistrationPage.jsx:304]: And here too.

**Review by waheedahmed — APPROVED** [review]
Just one suggestion otherwise looks good 👍

**Review by waheedahmed — APPROVED** [review]

**Review by adeelehsan — APPROVED** [review]
Looks good

**Pull request merged**